docs/api/networkinterfaces/MeshInterface.md

```diff
@@ -2,52 +2,29 @@
+The Arm Mbed Mesh API allows the application to use the IPv6 mesh network topologies through the [Nanostack](../reference/nanostack-introduction-tech.html) networking stack. 
 
 Mbed OS provides three types of IPv6 based mesh networks:
 
-- 6LoWPAN_ND, loosely following the Zigbee-IP specification.
-- Wi-SUN, following the specification from Wi-SUN alliance.
-- Thread, following the specification from Thread Group.
+- [Wi-SUN](../reference/wisun-tech.html), following the specification from the Wi-SUN alliance.
+- [Thread](../reference/thread-tech.html), following the specification from Thread Group.
+- [6LoWPAN-ND](../reference/6LoWPAN-ND-tech.html), loosely following the Zigbee-IP specification.
 
-Nanostack is the networking stack that provides these protocols. For more information on the stack internals, please refer to the [Thread](../reference/thread-tech.html), [Wi-SUN](../reference/wisun-tech.html) and [6LowPAN](../reference/mesh-tech.html) sections. Application developers use Nanostack through the Mbed Mesh API.
-
-The application can use the `LoWPANNDInterface`, `WisunInterface` or `ThreadInterface` object for connecting to the mesh network. When successfully connected, the application can use the Mbed [C++ socket APIs](network-socket.html) to create a socket to start communication with a remote peer.
+The application can use the `LoWPANNDInterface`, `WisunInterface` or `ThreadInterface` object for connecting to the mesh network. When successfully connected, the application can use the Mbed [C++ socket APIs](network-socket.html) to create a socket to start communication with a remote peer. You can use the [Network status API](network-status.html) to monitor changes in the network status.
 
 You can configure the mesh interface by providing values in `mbed_app.json`, as the [mesh configuration](../reference/configuration-mesh.html) section documents.
 
-## Usage
-
-1. Create a network interface and driver objects.
-1. Initialize the interface with given PHY driver.
-1. Connect to network.
-
-## Supported mesh networking modes
-
-Currently, 6LoWPAN-ND (neighbor discovery), Wi-SUN FAN and Thread bootstrap modes are supported.
-
-## Network connection states
-
-After the initialization, the network state is `MESH_DISCONNECTED`. After a successful connection, the state changes to `MESH_CONNECTED` and when disconnected from the network the state is changed back to `MESH_DISCONNECTED`.
-
-In case of connection errors, the state is changed to some of the connection error states. In an error state, there is no need to make a `disconnect` request and the application is allowed to attempt connecting again.
-
-## Getting started
-
-See the example application [mbed-os-example-mesh-minimal](https://github.com/ARMmbed/mbed-os-example-mesh-minimal) for usage.
-
 ## Mesh class reference
 
 [![View code](https://www.mbed.com/embed/?type=library)](https://os.mbed.com/docs/development/mbed-os-api-doxy/class_mesh_interface.html)
 
-## Mesh example
```

Comment:
Is there a brief, clever example we can transclude here?

Reply:
The https://github.com/ARMmbed/mbed-os-example-mesh-minimal/blob/master/main.cpp is the only one we have and I'm afraid it is not too brief. However, the application is referenced in the "Related content" section.

Reply:
Arto, we could include small 10 line snipped here, just to showcase that Mesh applications follow the same interface rules as any other interface.
Something like this:

# Mesh example

Following example connects to mesh network

```

#include "mbed.h"

int main(void)
{
	MeshInterface *mesh = MeshInterface::get_default_instance();

	int error = mesh->connect();
    if (error) {
        printf("Connection failed! error %d\n", error);
        return error;
    }

    printf("Connected!\n");

    UDPSocket sock;
    error = sock.open(mesh);
    if (error) {
    	printf("Failed to open socket, error %d\n", error);
    }

    // Now interface is connected, and I can communicate with Sockets
}
```

Reply:
Added in commit "Review corrections"